sql/analyzer/resolve_columns.go

```diff
@@ -239,27 +239,31 @@ func getColumnsInNodes(nodes []sql.Node, columns map[string][]string) {
 }
 
 func getNodeAvailableTables(n sql.Node) map[string]string {
-	var tables = make(map[string]string)
-	for _, c := range n.Children() {
-		plan.Inspect(c, func(n sql.Node) bool {
-			switch n := n.(type) {
-			case *plan.SubqueryAlias, *plan.ResolvedTable:
-				name := strings.ToLower(n.(sql.Nameable).Name())
+	tables := make(map[string]string)
+	getNodesAvailableTables(tables, n.Children()...)
+	return tables
+}
+
+func getNodesAvailableTables(tables map[string]string, nodes ...sql.Node) {
```

Comment:
maybe it can also return a tables map - WDYT?

Reply:
The reason why this does not return a table map is because it takes one as a parameter to add stuff to it and avoid having to merge maps since this is called recursively. It could return a map but it's pointless because it will return the same map as the one it receives.

Reply:
Makes sense. It's still tail recursion, so OK.